src/txadmission.cpp

```diff
@@ -0,0 +1,1028 @@
+// Copyright (c) 2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+//#include "chainparams.h"
+#include "blockstorage/blockstorage.h"
+#include "dosman.h"
+#include "init.h"
+#include "fastfilter.h"
+#include "net.h"
+#include "timedata.h"
+#include "txadmission.h"
+#include "txorphanpool.h"
+#include "consensus/tx_verify.h"
+#include "main.h" // for cs_main
+#include "respend/respenddetector.h"
+#include "unlimited.h"
+#include "util.h"
+#include "utiltime.h"
+#include "validationinterface.h"
+#include <map>
+#include <string>
+#include <vector>
+
+#include <boost/thread/thread.hpp>
+
+using namespace std;
+
+
+Snapshot txHandlerSnap;
+
+void ThreadCommitToMempool();
+void ThreadTxAdmission();
+void ProcessOrphans(std::vector<uint256> &vWorkQueue);
+
+bool CheckFinalTx(const CTransaction &tx, int flags, const Snapshot &ss);
+bool CheckSequenceLocks(const CTransaction &tx, int flags, LockPoints *lp, bool useExistingLockPoints, const Snapshot &ss);
+
+
+void StartTxAdmission(boost::thread_group &threadGroup)
+{
+    txHandlerSnap.Load(); // Get an initial view for the transaction processors
+
+    // Start incoming transaction processing threads
+    for (unsigned int i = 0; i < numTxAdmissionThreads.Value(); i++)
+    {
+        threadGroup.create_thread(boost::bind(&TraceThreads<void (*)()>, strprintf("tx%d", i), &ThreadTxAdmission));
+    }
+
+    // Start tx commitment thread
+    threadGroup.create_thread(boost::bind(&TraceThread<void (*)()>, "txcommit", &ThreadCommitToMempool));
+}
+
+void StopTxAdmission()
+{
+    cvTxInQ.notify_all();
+    cvCommitQ.notify_all();
+}
+
+void FlushTxAdmission()
+{
+    bool empty = false;
+
+    while (!empty)
+    {
+        do // give the tx processing threads a chance to run
+        {
+            {
+                LOCK(csTxInQ);
+                empty = txInQ.empty() & txDeferQ.empty();
+            }
+            if (!empty)
+                MilliSleep(100);
```

Comment:
why 100?

Reply:
no real reason, could be 200 or 50.  I want to give enough time for other threads to work before wasting context switch time again, but don't want to be too slow.